src/video_core/renderer_vulkan/renderer_vulkan.cpp

```diff
@@ -157,10 +160,38 @@ void RendererVulkan::RecreateFrame(Frame* frame, u32 width, u32 height) {
     frame->height = height;
 }
 
+bool RendererVulkan::ShowSplash(Frame* frame /*= nullptr*/) {
+    if (!Libraries::SystemService::IsSplashVisible()) {
+        return false;
+    }
+
+    if (!frame) {
+        const auto* splash = Common::Singleton<Splash>::Instance();
+
+        VideoCore::ImageInfo info{};
+        info.pixel_format = vk::Format::eR8G8B8A8Srgb;
+        info.type = vk::ImageType::e2D;
+        info.size =
+            VideoCore::Extent3D{splash->GetImageInfo().width, splash->GetImageInfo().height, 1};
+        info.pitch = splash->GetImageInfo().width * 4;
+        info.guest_size_bytes = splash->GetImageData().size();
+        auto& image = texture_cache.FindImage(info, VAddr(splash->GetImageData().data()));
```

Comment:
Using the texture cache also means registering the image, so using it should ideally be reserved for guest images. You may want to directly copy the image to the frame instead instead